### PR TITLE
feat: external real-time map url

### PR DESCRIPTION
**Pull request by mortennordseth**
Troms would like a link to their real-time map (https://kart.fylkestrafikk.no) from the map in the app. I've added the possibility to do so by adding a new button to the map, similar to the filter button, that is shown if the `externalRealtimeMap` URL is defined in the Firestore configuration. The button opens a bottom sheet with the URL. 

<details>
<summary>Screenshots</summary>

image

image

</details>

Closes https://github.com/AtB-AS/kundevendt/issues/18585

### CODE REVIEW

File: src/translations/components/Map.ts [hunk 1]
```diff
@@ -27,5 +28,29 @@ const MapTexts = {
       heading: _('Filter', 'Filter', 'Filter'),
     },
   },
+  externalRealtimeMap: {
+    bottomSheet: {
+      heading: _('Sanntidskart', 'Real-time map', 'Sanntidskart'),
```
Comment (adriansberg) [inline src/translations/components/Map.ts:33]: Hyphen is used here in the English translation, but not in the one below. Should be the same.

Comment (mortennordseth) [reply src/translations/components/Map.ts:33]: Fixed 👍

**Comment by mortennordseth** [comment]
> @mortennordseth when we have the filter button, how will the layout shows up?

Like this, @reidzeibel 

image

File: src/components/map/components/external-realtime-map/ExternalMapSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      title={t(MapTexts.externalRealtimeMap.bottomSheet.heading)}
+      onClose={onClose}
+    >
+      <View style={style.container}>
+        <ThemeText>
+          {t(MapTexts.externalRealtimeMap.bottomSheet.description)}
+        </ThemeText>
+      </View>
+      <FullScreenFooter>
+        <Button
```
Comment (reidzeibel) [inline src/components/map/components/external-realtime-map/ExternalMapSheet.tsx:33]: I think we need to add an accessibility part here:

```
              accessibilityHint={t(Hint.Message.Text)}
              accessibilityRole="link"
```

Comment (reidzeibel) [reply src/components/map/components/external-realtime-map/ExternalMapSheet.tsx:33]: Nevermind, this is a map screen, silly me....

**Review by reidzeibel — APPROVED** [review]

**Comment by rosvik** [comment]
Design suggestion: Could we re-use the look from the activate single ticket bottom sheet? That is, use a section item for the description

**Comment by mortennordseth** [comment]
> Design suggestion: Could we re-use the look from the activate single ticket bottom sheet? That is, use a section item for the description
> 

Good suggestion! ✅

**Review by rosvik — APPROVED** [review]
Looks good 👍

File: src/components/map/components/external-realtime-map/ExternalMapButton.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import {Button} from '@atb/components/button';
+import {StyleSheet} from '@atb/theme';
+import {Map} from '@atb/assets/svg/mono-icons/map';
+
+type ExternalMapButtonProps = {
+  onPress: () => void;
+};
+export const ExternalMapButton = ({onPress}: ExternalMapButtonProps) => {
```
Comment (tft-abel) [inline src/components/map/components/external-realtime-map/ExternalMapButton.tsx:9]: Naming might become confusing in the future: `ExternalMapButton` vs `ExternalRealtimeMap` in the Map.tsx file. We could align this to the Firestore Config and roll `External Realtime Map` everywhere (e..g `ExternalRealtimeMapButton` and `external-realtime-map`)

File: src/components/map/Map.tsx [hunk 4]
```diff
@@ -190,6 +195,11 @@ export const Map = (props: MapProps) => {
     ],
   );
 
+  const externalRealtimeMapUrl = getTextForLanguage(
+    configurableLinks?.externalRealtimeMap,
+    language,
+  );
```
Comment (gorandalum) [inline src/components/map/Map.tsx:201]: Also possible to encapsulate this into the `ExternalMapButton` component, by making it `return null` if no url in `configurableLinks`?

**Review by gorandalum — APPROVED** [review]

File: src/components/map/Map.tsx (hunk 4 above)
Comment (mortennordseth) [reply src/components/map/Map.tsx]: Sounds like a good change. I've updated to do so now

**Pull request merged**